ludwig/hyperopt/execution.py

```diff
@@ -447,10 +447,17 @@ def _run_experiment(
         trial_id = tune.get_trial_id()
         trial_dir = Path(tune.get_trial_dir())
 
+        # Write out the unmerged config to the trial's local directory.
+        with open(os.path.join(trial_dir, "user_hyperparameters.json"), "w") as f:
+            json.dump(hyperopt_dict["config"], f)
+
         modified_config = substitute_parameters(copy.deepcopy(hyperopt_dict["config"]), config)
 
         modified_config = merge_with_defaults(modified_config)
 
+        with open(os.path.join(trial_dir, "model_hyperparameters.json"), "w") as f:
+            json.dump(hyperopt_dict["config"], f)
```

Comment:
@jeffkinnison We actually write this during `model.train()` here: https://github.com/ludwig-ai/ludwig/blob/master/ludwig/api.py#L547

Is the reason to do this to get `model_hyperparameters.json` before we even get to `model.train()`? This would be useful is there's failures that may happen during preprocessing time since we only write `model_hyperparameters.json` after preprocessing.

Reply:
Would also encourage using `MODEL_HYPERPARAMETERS_FILE_NAME` from `ludwig.globals` just incase this file name changes :)

Reply:
Yeah, I see that we make some modifications to the config at the start of training. It probably makes more sense to read the rendered config on training start.

Reply:
This should be addressed by [7b40418](https://github.com/ludwig-ai/ludwig/pull/2640/commits/7b40418caeba34b8ab3c914984282f07542a75cf)